pkg/storegateway/threadpool/threadpool.go

```diff
@@ -21,45 +23,41 @@ const (
 var ErrPoolStopped = errors.New("thread pool has been stopped")
 
 type Threadpool struct {
+	services.Service
+
 	// pool is used for callers to acquire and return threads, blocking when they are all in use.
 	pool chan *osThread
 	// threads is used to perform operations on all threads at once (such as stopping and shutting down).
 	threads []*osThread
 	// stopping is closed when calling code wants the threadpool to shut down.
 	stopping chan struct{}
-	// stopped is closed once all threads have stopped running.
-	stopped chan struct{}
 
 	timing *prometheus.HistogramVec
 	tasks  prometheus.Gauge
 }
 
 // NewThreadPool creates a new instance of ThreadPool that runs tasks in a pool
 // of dedicated OS threads, meaning no other goroutines are scheduled on the threads.
-// If num is less than or equal to zero, nil is returned.
-func NewThreadPool(num int, reg prometheus.Registerer) *Threadpool {
-	if num <= 0 {
-		return nil
-	}
-
+// If num is zero, no threads will be started and all calls to Execute will run in
+// the goroutine of the caller.
+func NewThreadPool(num uint, reg prometheus.Registerer) *Threadpool {
```

Comment:
Nit: please rename to `Threatpool` to be consistent with the type name spelling.